pkg/apb/ext_creds.go

```diff
@@ -15,77 +16,56 @@ var timeoutFreq = 6    // Seconds
 var totalTimeout = 900 // 15min
 
 // ExtractCredentials - Extract credentials from pod in a certain namespace.
-// HACK ALERT!
-// A lot of the current approach to extracting credentials and monitoring
-// output is *very* experimental and error prone. Entire approach is going
-// to be thrown out and redone asap.
 func ExtractCredentials(
 	podname string, namespace string, log *logging.Logger,
 ) (*ExtractedCredentials, error) {
-	log.Debug("Calling monitorOutput on " + podname)
-	credOut, _ := monitorOutput(podname, namespace)
-	log.Debug("oc log output: %s", string(credOut))
+	credOut := make(chan []byte)
 
-	var creds *ExtractedCredentials
-	var err error
-	creds, err = buildExtractedCredentials(credOut)
+	log.Debug("Calling monitorOutput on " + podname)
+	go monitorOutput(podname, credOut, log)
```

Comment:
I'm okay with the goroutine here if it doesn't introduce complexity for maintenance purposes. Realistically, we don't need this here in the next few months, and in that time frame, we're definitely going to need to introduce async binds with the `WorkEngine`, so we'll have a strategic way to parallelize that work. Just wanted to comment so people are aware, this isn't needed when we spin off bind work.